crates/matrix-sdk-indexeddb/src/event_cache_store/indexeddb_serializer.rs

```diff
@@ -0,0 +1,489 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use base64::{
+    alphabet,
+    engine::{general_purpose, GeneralPurpose},
+    Engine,
+};
+use gloo_utils::format::JsValueSerdeExt;
+use matrix_sdk_crypto::CryptoStoreError;
+use matrix_sdk_store_encryption::{EncryptedValueBase64, StoreCipher};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use wasm_bindgen::JsValue;
+use web_sys::IdbKeyRange;
+use zeroize::Zeroizing;
+
+use crate::{safe_encode::SafeEncode, IndexeddbEventCacheStoreError};
+
+type Result<A, E = IndexeddbEventCacheStoreError> = std::result::Result<A, E>;
+
+const BASE64: GeneralPurpose = GeneralPurpose::new(&alphabet::STANDARD, general_purpose::NO_PAD);
+
+/// Handles the functionality of serializing and encrypting data for the
+/// indexeddb store.
+pub struct IndexeddbSerializer {
```

Comment:
Copy and paste from `crypto_store` not sure if a different serializer is needed or they can be merged and re-used by both modules.